test/build/cache/cache.test.js

```diff
@@ -218,4 +218,30 @@ describe("cache", () => {
     expect(stderr).toBeTruthy();
     expect(stdout).toBeTruthy();
   });
+
+  it("should gracefully exit when cache is true", (done) => {
+    fs.rmSync(path.join(__dirname, "../../../node_modules/.cache/webpack/graceful-exit-test"), {
+      recursive: true,
+      force: true,
+    });
+
+    const proc = runAndGetProcess(__dirname, [
+      "--config",
+      "./graceful-exit.webpack.config.js",
+      "--name",
+      "graceful-exit-test",
+    ]);
+    proc.stdout.on("data", (chunk) => {
+      const data = chunk.toString();
+      if (data.includes("app.bundle.js")) {
```

Comment:
after compile is done and we got an async plugin the sigterm is invoked while we're in the child process without an infinite loop fyi @alexander-akait